packages/viz/src/components/ConfusionMatrix/ConfusionMatrix.tsx

```diff
@@ -0,0 +1,299 @@
+import { useMemo } from "react";
+
+import {
+  Arrayable,
+  ExtractNames,
+  useTheme,
+} from "@hitachivantara/uikit-react-core";
+
+import * as echarts from "echarts/core";
+import { HeatmapChart } from "echarts/charts";
+import {
+  VisualMapComponent,
+  GridComponent,
+  TooltipComponent,
+} from "echarts/components";
+
+import {
+  useData,
+  useGrid,
+  useTooltip,
+  useVisualMap,
+  useXAxis,
+  useYAxis,
+} from "@viz/hooks";
+import { getGroupKey } from "@viz/utils";
+
+import { HvBaseChart } from "../BaseChart";
+import { useClasses } from "./ConfusionMatrix.styles";
+import {
+  HvChartCommonProps,
+  HvChartXAxis,
+  HvChartYAxis,
+} from "../../types/common";
+import { HvChartTooltip } from "../../types/tooltip";
+import { HvConfusionMatrixMeasure } from "../../types/measures";
+import { useColorScale, useGridLayout, useSeries } from "./utils";
+import {
+  HvConfusionMatrixFormat,
+  HvConfusionMatrixSemanticScale,
+  HvConfusionMatrixValuesProps,
+} from "./types";
+
+// Register chart components
+echarts.use([
+  HeatmapChart,
+  VisualMapComponent,
+  GridComponent,
+  TooltipComponent,
+]);
+
+export type HvConfusionMatrixClasses = ExtractNames<typeof useClasses>;
+
+export interface HvConfusionMatrixProps
+  extends Omit<HvChartCommonProps, "tooltip"> {
+  /** Column to measure. */
+  measure: HvConfusionMatrixMeasure;
+  /** Columns to use to split the measure. */
+  splitBy?: Arrayable<string>;
+  /**
+   * Column to use for the delta confusion matrix.
+   *
+   * It can be set to `true` in case the `measure` already has the calculations for the delta confusion matrix.
+   */
+  delta?: boolean | string;
+  /** Options for the xAxis, i.e. the horizontal axis. */
+  xAxis?: HvChartXAxis;
+  /** Options for the yAxis, i.e. the vertical axis. */
+  yAxis?: HvChartYAxis;
+  /** Tooltip options. */
+  tooltip?: Omit<HvChartTooltip, "type">;
+  /** Format of the confusion matrix. Defaults to `square`. */
+  format?: HvConfusionMatrixFormat;
+  /** Properties to customize the prediction values. */
+  valuesProps?: HvConfusionMatrixValuesProps;
+  /** A Jss Object used to override or extend the styles applied to the component. */
+  classes?: HvConfusionMatrixClasses;
+  /**
+   * Color scale of the confusion matrix. The first and second values are the lower and upper ends of the scale, respectively.
+   *
+   * The `colorScale` will be ignored when `semanticScale` is provided.
+   * If `delta` is not provided, a default color scale is used when `colorScale` is not defined: `[base-light, cat3]`.
+   */
+  colorScale?: [string, string];
+  /** Semantic scale used to distinguish between "good" and "bad" values. */
+  semanticScale?: HvConfusionMatrixSemanticScale[];
```

Comment:
Wouldn't it be simpler to merge these two into one, since only one of them will be applied and they kinda do the same thing? We would handle the differences internally.

```suggestion
  colorScale?: [string, string] | HvConfusionMatrixSemanticScale[];
```

Also, calling this `semantic` and "good" and "bad" values seems too specific (we also accept atmo colors). What if we want to plot political parties? Which is good, which is bad? 😈 😅
What do you think about `HvConfusionMatrixScale` or `HvConfusionMatrixColorScale`?